File: Java/RecyclerView17NetworkingWithAsyncHttpFuelVolleyOKHttpRetrofit/app/src/main/java/dev/ogabek/recyclerview17_networkingwithasynchttpfuelvolleyokhttpretrofit/activity/MainActivity.java
package dev.ogabek.recyclerview17_networkingwithasynchttpfuelvolleyokhttpretrofit.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.os.Bundle;
import android.widget.ProgressBar;

import java.util.List;

import dev.ogabek.recyclerview17_networkingwithasynchttpfuelvolleyokhttpretrofit.R;
import dev.ogabek.recyclerview17_networkingwithasynchttpfuelvolleyokhttpretrofit.adapter.MainAdapter;
import dev.ogabek.recyclerview17_networkingwithasynchttpfuelvolleyokhttpretrofit.internet.APIRequests;
import dev.ogabek.recyclerview17_networkingwithasynchttpfuelvolleyokhttpretrofit.model.Player;

public class MainActivity extends AppCompatActivity {

    Context context;
    RecyclerView recyclerView;
    ProgressBar progressBar;
    APIRequests apiRequests;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        apiRequests.apiAsyncHttp();

    }

    public void refleshAdapter(List<Player> players) {
        MainAdapter adapter = new MainAdapter(context, players);
        recyclerView.setAdapter(adapter);
    }

    private void initViews() {
        context = this;
        progressBar = findViewById(R.id.pb_mainProgressBar);
        apiRequests = new APIRequests(progressBar, MainActivity.this);
        recyclerView = findViewById(R.id.rv_mainRecyclerView);
        recyclerView.setLayoutManager(new GridLayoutManager(context, 1));
    }
}

File: Kotlin/RecyclerView17NetworkingWithAssycHttpFuelVolley/app/src/main/java/dev/ogabek/recyclerview_17_networkingwithassychttpfuelvolley/activity/MainActivity.kt
package dev.ogabek.recyclerview_17_networkingwithassychttpfuelvolley.activity

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ProgressBar
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.recyclerview_17_networkingwithassychttpfuelvolley.R
import dev.ogabek.recyclerview_17_networkingwithassychttpfuelvolley.adapter.MainAdapter
import dev.ogabek.recyclerview_17_networkingwithassychttpfuelvolley.internet.ApiRequests
import dev.ogabek.recyclerview_17_networkingwithassychttpfuelvolley.model.Player

class MainActivity : AppCompatActivity() {

    private lateinit var context: Context
    private lateinit var recyclerView: RecyclerView
    private lateinit var progressBar: ProgressBar
    private lateinit var apiRequests: ApiRequests

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

        apiRequests.apiAsyncHttp()

    }

    fun refleshAdapter(players: List<Player>) {
        val adapter = MainAdapter(context, players as ArrayList<Player>)
        recyclerView.adapter = adapter
    }

    private fun initViews() {
        context = this
        progressBar = findViewById(R.id.pb_main_progress_bar)
        apiRequests = ApiRequests(progressBar, this@MainActivity)
        recyclerView = findViewById(R.id.rv_main_recycler_view)
        recyclerView.layoutManager = GridLayoutManager(context, 1)
    }

}